### PR TITLE
feat: implement drag and drop filters controls

### PR DESCRIPTION
**Pull request by kgabryje**
🏆 Enhancements

Based on https://github.com/apache-superset/superset-ui/pull/967 by @zhaoyongjie 

CC: @villebro @junlincc

### CODE REVIEW

File: packages/superset-ui-chart-controls/src/shared-controls/dndControls.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  ...dndGroupByControl,
+  label: t('Entity'),
+  default: null,
+  multi: false,
+  validators: [validateNonEmpty],
+  description: t('This defines the element to be plotted on the chart'),
+};
+
+export const dnd_adhoc_filters: SharedControlConfig<'FilterColumnSelect'> = {
+  type: 'FilterColumnSelect',
```
Comment (zhaoyongjie) [inline packages/superset-ui-chart-controls/src/shared-controls/dndControls.tsx:75]: please use `dnd` as the control name prefix.

**Comment by zhaoyongjie** [comment]
I personally suggest not to make generic components now. The main reason is that these components are completely inconsistent in function, and we should not do such maintenance that increases complexity.

#### For dimension(group by) selector
1. columns (can shift between)
2. Level columns(AKA hierarchy) (there may be folding between levels, can not shift between)
3. [columns set](https://help.tableau.com/current/pro/desktop/en-us/sortgroup_sets_create.htm)(such as generate `case...when...` clause)

#### For filter selector, Superset is now just a simple dimension filter
1. If we want to drag saved_metric to generate a having clause,  we need to enable metic drop to filter
2. columns, or top level of hierarchy
3. Drag into different types of columns should have different filter

#### For metric selector
1. columns ( generate ad-hoc metric)
2. saved_metric

File: packages/superset-ui-chart-controls/src/shared-controls/dndControls.tsx (hunk 1 above)
Comment (kgabryje) [reply packages/superset-ui-chart-controls/src/shared-controls/dndControls.tsx]: Done, I renamed it to DndFilterSelect

**Review by villebro — APPROVED** [review]
LGTM

**Pull request merged**